mvpa2/measures/adhocsearchlightbase.py

```diff
@@ -555,12 +556,25 @@ def assign_ulabels(a):
                   (self.__class__.__name__, time.time() - time_start))
 
         out = Dataset(results)
-        if all_targets:
-            out.sa['targets'] = all_targets
+        self._all_testing_sa = all_testing_sa
         out.sa['cvfolds'] = all_cvfolds
         out.fa['center_ids'] = roi_ids
         return out
 
+    def _pass_attr(self, ds, result):
+        if self._all_testing_sa:
+            all_testing_sa = self._all_testing_sa
+            new_sas = {}
+            for attr in all_testing_sa[0].keys():
+                new_sas[attr] = np.concatenate([tsa[attr].value for tsa in all_testing_sa], 0)
```

Comment:
hm, so this one goes beyond of original pass_attr intention to just copy attributes from the original dataset, not to "track" them down all the way like this one does.  My concern is that it might be somewhat defeating the purpose of these adhoc searchlights to be fast -- in case of a dataset with heavy/lots of .sa's all the concatenation which would not get used (unless some pass_attr were specified) is somewhat wasteful :-/  At least condition it on having any 'self.pass_attr' defined and only then do this "copy everything" here and above... (otherwise -- only 'targets').  Actually ideally, at the top above (before looping) just call super's _pass_attr on an original ds, and mocked up result matching it to track which .sa would get copied, and then collect/concatenate only those to be passed along